src/input.rs

```diff
@@ -130,50 +132,50 @@ pub enum ButtonCode {
 
 impl ButtonCode {
     fn from_raw(code: u16, rotation: i8, button_scheme: ButtonScheme) -> ButtonCode {
-        if code == KEY_POWER {
-            ButtonCode::Power
-        } else if code == KEY_HOME {
-            ButtonCode::Home
-        } else if code == KEY_LIGHT {
-            ButtonCode::Light
-        } else if code == KEY_BACKWARD {
-            match rotation {
-                0 => ButtonCode::Backward,
-                1 => {
-                    match button_scheme {
-                        ButtonScheme::Natural => ButtonCode::Backward,
-                        ButtonScheme::Inverted => ButtonCode::Forward,
-                    }
-                },
-                2 => ButtonCode::Forward,
-                3 => {
-                    match button_scheme {
-                        ButtonScheme::Natural => ButtonCode::Forward,
-                        ButtonScheme::Inverted => ButtonCode::Backward,
-                    }
-                },
-                _ => ButtonCode::Backward,
+        match code {
+            KEY_POWER => ButtonCode::Power,
+            KEY_HOME => ButtonCode::Home,
+            KEY_LIGHT => ButtonCode::Light,
+            KEY_BACKWARD => resolve_button_direction(ButtonDirection::Backward, rotation, button_scheme),
+            KEY_FORWARD => resolve_button_direction(ButtonDirection::Forward, rotation, button_scheme),
+            _ => ButtonCode::Raw(code)
+        }
+    }
+}
+
+enum ButtonDirection {
+    Backward,
+    Forward
+}
+
+impl ButtonDirection {
+    fn flip(self) -> ButtonDirection {
```

Comment:
The name `flip` seems to indicate that the method mutates `self`. In addition, there's a `Dir::opposite` method in `geom.rs`.